recipes-kernel/linux/files/patches-5.10/0050-arm64-dts-ti-iot2050-Add-layout-of-OSPI-flash.patch

```diff
@@ -0,0 +1,71 @@
+From 4e3f7c215826106ca0eb369c7fb165cdb8d03b62 Mon Sep 17 00:00:00 2001
+From: Jan Kiszka <jan.kiszka@siemens.com>
+Date: Mon, 21 Mar 2022 15:53:15 +0100
+Subject: [PATCH 050/119] arm64: dts: ti: iot2050: Add layout of OSPI flash
+
+Describe the layout of the OSPI flash as the latest firmware uses it.
+Specifically the location of the U-Boot envs is important for userspace
+in order to access it.
+
+Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
+---
+ .../boot/dts/ti/k3-am65-iot2050-common.dtsi   | 43 ++++++++++++++++++-
+ 1 file changed, 41 insertions(+), 2 deletions(-)
+
+diff --git a/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi
+index 65da226847f4..7fa63b8ab8a6 100644
+--- a/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi
++++ b/arch/arm64/boot/dts/ti/k3-am65-iot2050-common.dtsi
+@@ -603,8 +603,47 @@ flash@0 {
+ 		cdns,tchsh-ns = <60>;
+ 		cdns,tslch-ns = <60>;
+ 		cdns,read-delay = <2>;
+-		#address-cells = <1>;
+-		#size-cells = <1>;
++
++		partitions {
++			compatible = "fixed-partitions";
++			#address-cells = <1>;
++			#size-cells = <1>;
++
++			seboot@0 {
++				label = "seboot";
++				reg = <0x0 0x180000>; /* 1.5M */
++			};
++
++			tispl@180000 {
++				label = "tispl";
++				reg = <0x180000 0x200000>; /* 2M */
++			};
++
++			u-boot@380000 {
++				label = "u-boot";
++				reg = <0x380000 0x300000>; /* 3M */
++			};
++
++			env@680000 {
++				label = "env";
++				reg = <0x680000 0x20000>; /* 128K */
++			};
++
++			env-backup@6a0000 {
++				label = "env.backup";
++				reg = <0x6a0000 0x20000>; /* 128K */
++			};
++
++			otpcmd@6c0000 {
++				label = "otpcmd";
++				reg = <0x6c0000 0x10000>; /* 64K */
++			};
++
++			free@6d0000 {
++				label = "free";
++				reg = <0x6d0000 0x930000>;
++			};
```

Comment:
Will there be a seperate mtdX partition mapping to this `free` partition?

BTW, I think it's the good time to add the backup seboot partition here, starting from 14.5MB to the end.

Reply:
This is what you will get:
```
root@iot2050-debian:~# cat /proc/mtd 
dev:    size   erasesize  name
mtd0: 00180000 00010000 "seboot"
mtd1: 00200000 00010000 "tispl"
mtd2: 00300000 00010000 "u-boot"
mtd3: 00020000 00010000 "env"
mtd4: 00020000 00010000 "env.backup"
mtd5: 00010000 00010000 "otpcmd"
mtd6: 00930000 00010000 "free"
```

Yes, we can reserve that backup partition as well. How should it be called?

Reply:
`seboot.reserved` or `seboot.b` ?

Reply:
or following the `env.backup`: `seboot.backup`